Automattic/woocommerce-payments
# Transactions: Add 'Payment' method filter to the advanced filters dropdown

Fixes #2749 

#### Changes proposed in this Pull Request

This PR adds the payment options to the type dropdown in the advanced filters of the Transactions page. The options added to the dropdown are:

- Payment
- Payment refund
- Payment failure refund

There was a TODO in the code, with this linked [issue](https://github.com/Automattic/woocommerce-payments/issues/692), to add the Payment options once we have support for additional payment methods. My understanding is that it is safe to do it now. 

**Before**
![Screen Capture on 2021-10-01 at 17-40-37](https://user-images.githubusercontent.com/1553182/135652993-ace07070-9c5f-4086-9c0a-6446688c3f94.gif)

#### Testing instructions
**Prerequisites**
1. Enable additional payment methods in WooCommerce > Settings > Payments. 
2. Create some orders selecting an additional payment method in the checkout and credit card. You will use those orders for testing. Hint: If you don't see the additional payment methods in the checkout page, try changing the customer's country i.e. Netherlands

**Testing**
1. Go to WooCommerce > Payments > Transactions
2. Check that the Transactions table shows all your transactions
3. Open Show: All Transactions dropdown
4. Select "Advanced Filters"
5. Click on "Add Filter"
6. Select "Type" in the dropdown
7. Click on the "Type Is" dropdown (it shows "Charge" option by default)
8. Select any "Payment option": "Payment", "Payment refund", "Payment failure refund"
9. Click on "Filter"
10. Check that the table shows transactions of the type you chose

This is a gif of the behavior after the fix:

![Screen Capture on 2021-10-01 at 17-44-15](https://user-images.githubusercontent.com/1553182/135655846-b569ea35-dc5b-4c23-b7ff-c405fca12641.gif)


-------------------


**Post merge**
